### PR TITLE
Added validation rules when creating CellOrder record

### PR DESCRIPTION
**Pull request by Roman45321**
Added validation rules:
- "Disallow_populating_Cell_field" - prohibits selecting a cell when the order status is "Pending delivery" or "Delivering", only Parcel Locker.
- "Require_Cell_for_delivered_status" - prohibits the absence of a value in the Cell field when the order status is "Delivered" or "Completed".
- "Ensure_Cell_matches_parcel_locker" - checks if the cell matches the selected Parcel Locker.
- "Lock_fields_after_delivery_completed" - prohibits editing fields other than status, when the order status is "Delivered" or "Completed".

### CODE REVIEW

File: force-app/main/default/objects/CellOrder__c/fields/Order__c.field-meta.xml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+<CustomField xmlns="http://soap.sforce.com/2006/04/metadata">
+    <fullName>Order__c</fullName>
+    <deleteConstraint>Restrict</deleteConstraint>
+    <label>Order</label>
+    <referenceTo>Order</referenceTo>
+    <relationshipLabel>Cell Orders</relationshipLabel>
+    <relationshipName>Cell_Orders</relationshipName>
+    <required>true</required>
+    <trackTrending>false</trackTrending>
+    <type>Lookup</type>
```
Comment (msenyk) [inline force-app/main/default/objects/CellOrder__c/fields/Order__c.field-meta.xml:11]: It should be Mater-Detail type.

File: force-app/main/default/objects/CellOrder__c/validationRules/Lock_fields_after_delivery_completed.validationRule-meta.xml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<ValidationRule xmlns="http://soap.sforce.com/2006/04/metadata">
+    <fullName>Lock_fields_after_delivery_completed</fullName>
+    <active>true</active>
+    <errorConditionFormula>AND(
+    OR(
+        AND(ISPICKVAL(PRIORVALUE(State__c), &apos;Delivered&apos;), ISPICKVAL(State__c, &apos;Completed&apos;)),
+        AND(ISPICKVAL(PRIORVALUE(State__c), &apos;Completed&apos;), ISPICKVAL(State__c, &apos;Delivered&apos;)),
+        AND(ISPICKVAL(PRIORVALUE(State__c), &apos;Delivered&apos;), ISPICKVAL(State__c, &apos;Delivered&apos;)),
+        AND(ISPICKVAL(PRIORVALUE(State__c), &apos;Completed&apos;), ISPICKVAL(State__c, &apos;Completed&apos;))
```
Comment (msenyk) [inline force-app/main/default/objects/CellOrder__c/validationRules/Lock_fields_after_delivery_completed.validationRule-meta.xml:10]: It is too complicated logic. We can check only prior value equals to "Delivered" and "Completed".

**Review by msenyk — CHANGES_REQUESTED** [review]

**Comment by Roman45321** [comment]
Removed unnecessary validations, changed Order field type to Master-Detail

**Comment by Roman45321** [comment]
Resolved conflicts with main branch in Order__c.field-meta.xml

**Pull request merged**